plugins/algoltek-usb/fu-algoltek-usb-device.c

```diff
@@ -0,0 +1,514 @@
+					   FU_ALGOLTEK_CMD_RST,
+					   0,
+					   0,
+					   buf,
+					   buf->data[0],
+					   error)) {
+		g_prefix_error(error, "system reboot failure: ");
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+static gboolean
+algoltek_device_wrr(FuAlgoltekUsbDevice *self, int address, int value, GError **error)
+{
+	g_autoptr(GByteArray) buf = fu_struct_algoltek_cmd_address_pkt_new();
+
+	fu_struct_algoltek_cmd_address_pkt_set_len(buf, 7);
+	fu_struct_algoltek_cmd_address_pkt_set_cmd(buf, FU_ALGOLTEK_CMD_WRR);
+	fu_struct_algoltek_cmd_address_pkt_set_address(buf, address);
+	fu_struct_algoltek_cmd_address_pkt_set_value(buf, value);
+	fu_struct_algoltek_cmd_address_pkt_set_checksum(buf, ~fu_sum8(buf->data, buf->len) + 1);
+
+	if (!algoltek_device_ctrl_transfer(self,
+					   G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+					   FU_ALGOLTEK_CMD_WRR,
+					   0,
+					   0,
+					   buf,
+					   buf->data[0],
+					   error)) {
+		g_prefix_error(error, "data write failure: ");
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+static gboolean
+algoltek_device_isp(FuAlgoltekUsbDevice *self,
+		    GBytes *blob_isp,
+		    guint address,
+		    FuProgress *progress,
+		    GError **error)
+{
+	const guint8 *isp_data = NULL;
+	gsize isp_data_size = 0;
+	guint8 basic_data_size = 5;
+	g_autoptr(GPtrArray) chunks_isp = NULL;
+
+	isp_data = g_bytes_get_data(blob_isp, &isp_data_size);
+	chunks_isp =
+	    fu_chunk_array_new(isp_data, isp_data_size, address, 0x0, 64 - basic_data_size);
+
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_set_steps(progress, chunks_isp->len);
+
+	for (guint i = 0; i < chunks_isp->len; i++) {
+		FuChunk *chk = g_ptr_array_index(chunks_isp, i);
+		g_autoptr(GByteArray) part_isp_data = fu_struct_algoltek_cmd_transfer_pkt_new();
+
+		fu_struct_algoltek_cmd_transfer_pkt_set_len(part_isp_data,
+							    basic_data_size +
+								fu_chunk_get_data_sz(chk));
+		fu_struct_algoltek_cmd_transfer_pkt_set_cmd(part_isp_data, FU_ALGOLTEK_CMD_ISP);
+		fu_struct_algoltek_cmd_transfer_pkt_set_address(part_isp_data,
+								fu_chunk_get_address(chk));
+		if (!fu_struct_algoltek_cmd_transfer_pkt_set_data(part_isp_data,
+								  fu_chunk_get_data(chk),
+								  fu_chunk_get_data_sz(chk),
+								  error)) {
+			g_prefix_error(error, "assign isp data failure: ");
+			return FALSE;
+		}
+		fu_struct_algoltek_cmd_transfer_pkt_set_checksum(
+		    part_isp_data,
+		    ~fu_sum8(part_isp_data->data, part_isp_data->len) + 1);
+
+		if (!algoltek_device_ctrl_transfer(self,
+						   G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+						   FU_ALGOLTEK_CMD_ISP,
+						   0,
+						   0,
+						   part_isp_data,
+						   part_isp_data->data[0],
+						   error)) {
+			g_prefix_error(error, "isp failure: ");
+			return FALSE;
+		}
+		fu_progress_step_done(progress);
+	}
+
+	return TRUE;
+}
+
+static gboolean
+algoltek_device_bot(FuAlgoltekUsbDevice *self, int address, GError **error)
+{
+	g_autoptr(GByteArray) buf = fu_struct_algoltek_cmd_address_pkt_new();
```

Comment:
usually, by convention, a suggestion -- we normally use `st` as the variable name for the thing created by `fu_struct_whatever` -- `buf` is normally the bare buffer we pass to things.

Reply:
Updated! Revise the variable name to `buf` and `st`, depending on whether it was created by  `fu_struct_whatever` .